packages/amplify-graphql-model-transformer/src/resources/dynamo-model-resource-generator.ts

```diff
@@ -0,0 +1,301 @@
+import * as cdk from '@aws-cdk/core';
+import { TransformerContextProvider } from '@aws-amplify/graphql-transformer-interfaces';
+import {
+  ModelResourceIDs,
+  ResourceConstants,
+  SyncResourceIDs,
+} from 'graphql-transformer-common';
+import { ObjectTypeDefinitionNode } from 'graphql';
+import { SyncUtils, TransformerNestedStack } from '@aws-amplify/graphql-transformer-core';
+import {
+  AttributeType,
+  CfnTable,
+  StreamViewType,
+  Table,
+  TableEncryption,
+} from '@aws-cdk/aws-dynamodb';
+import * as iam from '@aws-cdk/aws-iam';
+import { CfnDataSource } from '@aws-cdk/aws-appsync';
+import { CfnRole } from '@aws-cdk/aws-iam';
+import { ModelResourceGenerator } from './model-resource-generator';
+import { DynamoDBModelVTLGenerator, ModelVTLGenerator } from '../resolvers';
+
+/**
+ * DynamoModelResourceGenerator is an implementation of ModelResourceGenerator,
+ * providing necessary utilities to generate the DynamoDB resources for models
+ */
+export class DynamoModelResourceGenerator extends ModelResourceGenerator {
+  protected generatorType = 'DynamoModelResourceGenerator';
+
+  generateResources(ctx: TransformerContextProvider): void {
+    if (!this.isEnabled()) {
+      return;
+    }
+
+    if (this.isProvisioned()) {
+      // add model related-parameters to the root stack
+      ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBModelTableReadIOPS, {
+        description: 'The number of read IOPS the table should support.',
+        type: 'Number',
+        default: 5,
+      });
+      ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBModelTableWriteIOPS, {
+        description: 'The number of write IOPS the table should support.',
+        type: 'Number',
+        default: 5,
+      });
+      ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBBillingMode, {
+        description: 'Configure @model types to create DynamoDB tables with PAY_PER_REQUEST or PROVISIONED billing modes.',
+        default: 'PAY_PER_REQUEST',
+        allowedValues: ['PAY_PER_REQUEST', 'PROVISIONED'],
+      });
+      ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBEnablePointInTimeRecovery, {
+        description: 'Whether to enable Point in Time Recovery on the table.',
+        type: 'String',
+        default: 'false',
+        allowedValues: ['true', 'false'],
+      });
+      ctx.stackManager.addParameter(ResourceConstants.PARAMETERS.DynamoDBEnableServerSideEncryption, {
+        description: 'Enable server side encryption powered by KMS.',
+        type: 'String',
+        default: 'true',
+        allowedValues: ['true', 'false'],
+      });
+    }
+
+    this.models.forEach((model) => {
+      // This name is used by the mock functionality. Changing this can break mock.
+      const tableBaseName = ctx.resourceHelper.getModelNameMapping(model!.name.value);
```

Comment:
Do we need the `!` here? Is there a way to fallback to something if the model is undefined?

Reply:
Does `this.models` refer to DynamoDB models only? How do we filter it?

Reply:
it's filtered during the process of adding to the resource generators, the `!` is a holdover from copying the existing code in the model transformer, I can probably delete it